### PR TITLE
Embed image URLs

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -121,13 +125,32 @@
...
+    var ins;
+
+    if (probable_picture(m[1])) {
+      ins = '<img class="embedded" src="' + encodeURI(m[1]) + '">';
+    } else {
+      ins = '<a href="' + encodeURI(m[1]) + '">' + m[1] + '</a>';
+    }
     text = text.substring(0, i + m.index) + ins +
-      text.substring(i + m.index + m[1].length);
+    text.substring(i + m.index + m[1].length);
```
Comment (bifurcation) [inline index.html]: What happened to the indent?

Comment (ekr) [inline index.html]: I got confused.

**Comment by ekr** [comment]
@bifurcation fixed

File: index.html [hunk 2]
```diff
@@ -121,11 +125,30 @@
...
+
 function clean(text) {
   var i = 0, m;
   text = text.replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
   while((m = /\b(https?:\/\/\S+)/.exec(text.substring(i)))) {
-    var ins = '<a href="' + encodeURI(m[1]) + '">' + m[1] + '</a>';
+    var ins;
+
+    if (probable_picture(m[1])) {
+      ins = '<img class="embedded" src="' + encodeURI(m[1]) + '">';
```
Comment (martinthomson) [inline index.html:148]: Maybe link to the image.  Also note that you are creating a mixed content mess here.  Do we need CSP?

Comment (ekr) [inline index.html:148]: You mean wrap it with a link? I can do that. This is just passive mixed content, though.

**Pull request merged**